pkg/skaffold/deploy/docker/deploy.go

```diff
@@ -138,23 +151,68 @@ func (d *Deployer) deploy(ctx context.Context, out io.Writer, b graph.Artifact)
 		return err
 	}
 
-	containerName := d.getContainerName(ctx, b.ImageName)
+	containerCfg, err := d.containerConfigFromImage(ctx, b.Tag)
+	if err != nil {
+		return err
+	}
+	containerCfg.ExposedPorts = ports
+
+	initContainers, err := d.debugger.TransformImage(ctx, b, containerCfg)
+	if err != nil {
+		return errors.Wrap(err, "transforming image for debugging")
+	}
+
+	for _, c := range initContainers {
+		if d.debugger.HasMount(c.Image) {
+			// skip duplication of init containers
+			continue
+		}
+		if err := d.client.Pull(ctx, out, c.Image); err != nil {
+			return errors.Wrap(err, "pulling init container image")
+		}
+		id, err := d.client.Run(ctx, out, c, dockerutil.ContainerCreateOpts{})
+		if err != nil {
+			return errors.Wrap(err, "creating container in local docker")
+		}
+		r, err := d.client.ContainerInspect(ctx, id)
+		if err != nil {
+			return errors.Wrap(err, "inspecting init container")
+		}
+		if len(r.Mounts) != 1 {
+			olog.Entry(ctx).Warnf("unable to retrieve mount from debug init container: debugging may not work correctly!")
+		}
+		d.debugger.AddSupportMount(c.Image, r.Mounts[0].Name)
+	}
 
+	containerName := d.getContainerName(ctx, b.ImageName)
+	var mounts []mount.Mount
+	for _, m := range d.debugger.SupportMounts() {
+		mounts = append(mounts, m)
+	}
 	opts := dockerutil.ContainerCreateOpts{
```

Comment:
Rather than pass the container config as an argument, perhaps it could be included here?

Reply:
good idea